### PR TITLE
fixed: macCatalyst

### LINKED ISSUE #46: Mac Catalyst Login Item
Hello! Has anyone implemented this project with a Mac Catalyst app so far?

Would be grateful for any tips!

### PR DESCRIPTION
**Pull request by plananderson**
Added support for macCatalyst

Fixes #46

### CODE REVIEW

File: Sources/LaunchAtLogin/LaunchAtLogin.swift [hunk 1]
```diff
@@ -31,14 +31,25 @@ public enum LaunchAtLogin {
 			}
 
 			kvo.willChangeValue(for: \.isEnabled)
-			SMLoginItemSetEnabled(id as CFString, newValue)
+			Self._SMLoginItemSetEnabled(id: id, isEnabled: newValue)
```
Comment (sindresorhus) [inline Sources/LaunchAtLogin/LaunchAtLogin.swift:34]: This method should only be used on Catalyst. Use a `#if` check.

File: Sources/LaunchAtLogin/LaunchAtLogin.swift [hunk 2]
```diff
@@ -31,14 +31,25 @@ public enum LaunchAtLogin {
...
 			kvo.didChangeValue(for: \.isEnabled)
 
 			if #available(macOS 10.15, *) {
 				_publisher.send(newValue)
 			}
 		}
 	}
+	
+	private static func _SMLoginItemSetEnabled(id: String, isEnabled: Bool) {
```
Comment (sindresorhus) [inline Sources/LaunchAtLogin/LaunchAtLogin.swift:43]: This should be wrapped in an `#if` check for Catalyst. I would also move it to a private top-level function.

File: Sources/LaunchAtLogin/LaunchAtLogin.swift [hunk 3]
```diff
@@ -31,14 +31,25 @@ public enum LaunchAtLogin {
...
+		let handle = dlopen(nil, RTLD_LAZY)
+		let sym = dlsym(handle, "SMLoginItemSetEnabled")
```
Comment (sindresorhus) [inline Sources/LaunchAtLogin/LaunchAtLogin.swift:45]: Not clear to me what's going on here. A code comment would be useful to explain how exactly this makes the macOS API available in Catalyst.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Mention in the readme that Catalyst is supported.

**Comment by sindresorhus** [comment]
Have you tried this solution in a shipping app? Will Apple allow this on the App Store?

**Comment by frogg** [comment]
I am using LaunchAtLogin in Homie which is a Catalyst app: https://apps.apple.com/app/apple-store/id1533590432

**Pull request closed without merging**